tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts

```diff
@@ -19,24 +20,122 @@ class SeoTab extends BOBasePage {
 
   private readonly productRedirectProduct: string;
 
+  private readonly tagInput: string;
+
+  private readonly alertText: string;
+
+  private readonly generateURLFromNameButton: string;
+
+  private readonly redirectionWhenOfflineSelect: string;
+
+  private readonly searchOptionTargetInput: string;
+
   /**
    * @constructs
    * Setting up texts and selectors to use on Shipping tab
    */
   constructor() {
     super();
 
-    // Selectors in stocks tab
+    // Selectors in seo tab
     this.productMetaTitleInput = '#product_seo_meta_title';
     this.productMetaDescriptionInput = '#product_seo_meta_description';
     this.productLinkRewriteInput = '#product_seo_link_rewrite';
     this.productRedirectTypeSelect = '#product_seo_redirect_option_type';
     this.productRedirectProduct = '#product_seo_redirect_option_target_0_id';
+    this.tagInput = '#product_seo_tags_1-tokenfield';
+    this.alertText = '#product_seo div.alert-danger div.alert-text';
+    this.generateURLFromNameButton = '#product_seo button.reset-link-rewrite';
+    this.redirectionWhenOfflineSelect = '#product_seo_redirect_option_type';
+    this.searchOptionTargetInput = '#product_seo_redirect_option_target_search_input';
   }
 
   /*
   Methods
    */
+  /**
+   * Set meta title
+   * @param page {Page} Browser tab
+   * @param metaTitle {string} Meta title to set in the input
+   * @returns {Promise<void>}
+   */
+  async setMetaTitle(page: Page, metaTitle: string): Promise<void> {
+    await this.setValue(page, `${this.productMetaTitleInput}_1`, metaTitle);
+  }
+
+  /**
+   * Set meta description
+   * @param page {Page} Browser tab
+   * @param metaDescription {string} Meta description to set in the input
+   * @returns {Promise<void>}
+   */
+  async setMetaDescription(page: Page, metaDescription: string): Promise<void> {
+    await this.setValue(page, `${this.productMetaDescriptionInput}_1`, metaDescription);
+  }
+
+  /**
+   * Set friendly URL
+   * @param page {Page} Browser tab
+   * @param friendlyUrl {string} Friendly URL to set in the input
+   * @returns {Promise<void>}
+   */
+  async setFriendlyUrl(page: Page, friendlyUrl: string): Promise<void> {
+    await this.setValue(page, `${this.productLinkRewriteInput}_1`, friendlyUrl);
+  }
+
+  /**
+   * Get error message of friendly URL
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  async getErrorMessageOfFriendlyUrl(page: Page): Promise<string> {
+    await this.clickAndWaitForLoadState(page, createProductPage.saveProductButton);
+
+    return this.getTextContent(page, this.alertText);
+  }
```

Comment:
Maybe be having this in the addProductPage instead as is not linked to the this tab only

Reply:
no it's an error alert related to friendlyUrl section

![Capture d’écran de 2023-10-05 16-00-30](https://github.com/PrestaShop/PrestaShop/assets/24647495/dea6c102-dc3f-458e-835b-2fe66fe84547)

Reply:
```suggestion
  /**
   * Get error message of friendly URL
   * @param page {Page} Browser tab
   * @returns {Promise<string>}
   */
  async getErrorMessageOfFriendlyUrl(page: Page): Promise<string> {
    return this.getTextContent(page, this.alertText);
  }
```

Maybe like this and do 2 actions on the test 1 for save the product the second to get the message

Reply:
Done thx